### ISSUE TITLE
Add -c option for alternative config file

### ISSUE THREAD

**Issue by omangin** [pubs/pubs#19]

**Comment by 73** [comment]
Hi,
I have some code on this one but it seems that I can't get my head around the FakeFilesystem for testing. 
The statement:
```python
self.fs['os'].path.exists(self.fs['os'].path.expanduser('~/.pubsrc'))
```
always returns `false`. 
The users home should be initialized https://github.com/pubs/pubs/blob/master/tests/fake_env.py#L116 right? 
But the config file does not seem to reside inside the fake file system. 
To recreate just add the line to a test with an `pubs init`.

What am I missing here?
--73

**Comment by humm** [comment]
Yes, FakeFilesystem is kinda tricky. 

If you modified the code in ways that add calls to the filesystem in a new module, you must add it to:
https://github.com/pubs/pubs/blob/master/tests/test_usecase.py#L59
and similar lines in  https://github.com/pubs/pubs/blob/master/tests/test_databroker.py

Frankly, it needs to be more robust to changes, I'll add that to the todo list. 

If that does not solve your problem, I need to see your code to help you more.

**Comment by 73** [comment]
Well, here it is: https://github.com/73/pubs/commit/2e0fca5430c721a22c42e151f41c27fca890522d

The configs are already added. `open()` should be called inside the `configobj` that might be the issue, but I'm really not sure.  

Thanks,
--73

**Comment by humm** [comment]
It appear you have brought old code along with your changes when rebasing against the last commit of the `feat/new_config` branch. Your problems should be solved by a cleaner patch. The `feat/new_config` branch is almost finished, but wide-ranging color changes, and a merge with the new master code will still make it a highly moving target for a day or two.

**Comment by 73** [comment]
Thx for the feedback, I'll clean up.

**Comment by humm** [comment]
Ok, I pushed my color changes and merged with master. The code should be relatively stable now. It will be pushed into master soon.

**Comment by 73** [comment]
Nice. I ran into a problem which I can't decide for myself. 
Currently I can't call `config.get_confpath()` to determine the correct order of precedence for the config file (which I guess would be switch > env > default), because the function relies on `uis.get_ui()` https://github.com/pubs/pubs/blob/feat/new_config/pubs/config/conf.py#L30. But I need a config file to create it https://github.com/pubs/pubs/blob/feat/new_config/pubs/uis.py#L31-L39. Therefore there is an dependency cycle. From the top of my head I see to options to resolve this:
1. `uis` depends on the `conf`, not the other way around. Formatting must be handled outside the config.
2.  They are indeed dependent on each other and I may create a `default_config` to fall back on.

**Comment by humm** [comment]
Yep, you should load the default config if you need the ui before the configuration is loaded. 
You can get it there:
https://github.com/pubs/pubs/blob/feat/new_config/pubs/config/conf.py#L13

The ui will eventually become an argument passed along the function (same as conf and args), the current design is not pretty.

**Comment by 73** [comment]
@humm could you review this: https://github.com/73/pubs/commit/b4f569dee86c37534bd626f655ed040d7ad01fce
Should be a lot more pleasing to the eye ;)

**Comment by humm** [comment]
Fixed by 2871588
